test-runner-poc/lib/filters/mongodb_version_filter.js

```diff
@@ -14,67 +14,28 @@ const f = require('util').format;
  * }
  */
 class MongoDBVersionFilter {
-  constructor(options) {
-    this.options = options || {};
-    this.version = null;
-  }
-
-  beforeStart(configuration, callback) {
-    const self = this;
-    if (this.options.skip) {
-      return callback();
-    }
-
-    if (configuration.type === 'core') {
-      configuration.newConnection({ w: 1 }, function(err, topology) {
-        if (err) {
-          callback(err);
-          return;
-        }
-
-        topology.command(f('%s.$cmd', configuration.db), { buildInfo: true }, function(
-          commandErr,
-          result
-        ) {
-          if (commandErr) throw commandErr;
-          self.version = result.result.version;
-          console.log('running against mongodb version:');
-          console.dir(result.result);
-
-          topology.destroy();
-          callback();
-        });
+  initializeFilter(callback) {
+    const mongoClient = new MongoClient(process.env.MONGODB_URI || 'mongodb://127.0.0.1:27017');
+    mongoClient.connect((err, client) => {
+      if (err) throw new Error(err);
+      client.db('admin').command({ buildInfo: true }, (err, result) => {
+        if (err) throw new Error(err);
+        this.mongoVersion = result.version;
+        client.close(callback);
       });
-    } else {
-      configuration.newClient({ w: 1 }).connect(function(err, client) {
-        if (err) {
-          callback(err);
-          return;
-        }
-
-        client.db('admin').command({ buildInfo: true }, function(_err, result) {
-          if (_err) {
-            callback(_err);
-            return;
-          }
-
-          self.version = result.versionArray.slice(0, 3).join('.');
-          console.log('running against mongodb version:');
-          console.dir(result);
+    });
+  }
 
-          client.close();
-          callback();
-        });
-      });
-    }
+  constructor() {
+    this.options = {};
   }
 
   filter(test) {
     if (this.options.skip) return true;
```

Comment:
How will `this.options.skip` be true if we're setting `this.options = {}` in the [constructor](https://github.com/mongodb/node-mongodb-native/pull/2085/files#diff-f21fddb92f2e2d102b4b81c5fe5a1095R30)?

Reply:
Removed the constructor and `this.options.skip()`

Reply:
Do we still need some sort of `this.options.skip` logic or is it okay to leave that out?

Reply:
updated to get rid of `this.options.skip` logic!